### DISCUSSION TITLE
Error run fastapi-mqtt using gunicorn

### DISCUSSION THREAD

**Discussion by ruem2802** [fastapi/fastapi#8836] [Questions]
### First Check

- [x] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi_mqtt.fastmqtt import FastMQTT
from fastapi import FastAPI
from fastapi_mqtt.config import MQQTConfig

mqtt_config = MQQTConfig()

fast_mqtt = FastMQTT(
    config=mqtt_config
)


app = FastAPI()

fast_mqtt.init_app(app)


@fast_mqtt.on_connect()
def connect(client, flags, rc, properties):
    fast_mqtt.client.subscribe("/mqtt") #subscribing mqtt topic 
    print("Connected: ", client, flags, rc, properties)

@fast_mqtt.subscribe("mqtt/+/temperature", "mqtt/+/humidity")
async def home_message(client, topic, payload, qos, properties):
    print("temperature/humidity: ", topic, payload.decode(), qos, properties)
    return 0

@fast_mqtt.on_message()
async def message(client, topic, payload, qos, properties):
    print("Received message: ",topic, payload.decode(), qos, properties)
    return 0

@fast_mqtt.on_disconnect()
def disconnect(client, packet, exc=None):
    print("Disconnected")

@fast_mqtt.on_subscribe()
def subscribe(client, mid, qos, properties):
    print("subscribed", client, mid, qos, properties)

@app.get("/")
async def func():
    fast_mqtt.publish("/mqtt", "Hello from Fastapi") #publishing mqtt topic 

    return {"result": True,"message":"Published" }
```

### Description

I use fastapi-mqtt==0.3.0, python==3.9.7, fastapi==0.68.1
I run example application in fastapi-mqtt: https://github.com/sabuhish/fastapi-mqtt/blob/master/examples/app.py
When I run the app using uvicorn : uvicorn app:app --host 0.0.0.0 --port 7000 --reload
Everything is ok -> app run

But:
I run example application using command :
gunicorn app:app -k uvicorn.workers.UvicornWorker --workers=9 -b 0.0.0.0:7000

Error :
File "/home/hoanganh/smarthome-server/venv/lib/python3.9/site-packages/gmqtt/client.py", line 230, in connect
await self._connected.wait()
File "/usr/lib/python3.9/asyncio/locks.py", line 226, in wait
await fut
RuntimeError: Task <Task pending name='Task-3' coro=<LifespanOn.main() running at /home/hoanganh/smarthome-server/venv/lib/python3.9/site-packages/uvicorn/lifespan/on.py:84>> got Future attached to a different loop

[2021-09-21 20:42:13 +0700] [10419] [ERROR] Application startup failed. Exiting.
[2021-09-21 20:42:13 +0700] [10419] [INFO] Worker exiting (pid: 10419)
Task was destroyed but it is pending!
task: <Task pending name='Task-1' coro=<Client._resend_qos_messages() running at /home/hoanganh/smarthome-server/venv/lib/python3.9/site-packages/gmqtt/client.py:176>>
sys:1: RuntimeWarning: coroutine 'Client._resend_qos_messages' was never awaited
[2021-09-21 20:42:14 +0700] [10418] [INFO] Shutting down: Master
[2021-09-21 20:42:14 +0700] [10418] [INFO] Reason: Worker failed to boot.

### Operating System

Linux

### FastAPI Version

0.68.1

### Python Version

3.9.7

**Comment by Kludex** [comment]
This is both `gmqtt` and `fastapi-mqtt` fault. It's the same issue we have with `motor`.

Step by step:
1. The example initializes `FastMQTT`.
2. On `FastMQTT`, they [initialize the `MQTTClient`](https://github.com/sabuhish/fastapi-mqtt/blob/5ff5e26dd901458135276c300c81156d51fcee9c/fastapi_mqtt/fastmqtt.py#L58).
3. `MQTTClient` is a class that inherits some other classes on the `gmqtt` project, and one of those classes is [`EventCallback`](https://github.com/wialon/gmqtt/blob/50bc08300d858c6e28d4975a989be578954bf7bd/gmqtt/mqtt/handler.py#L67).
4. `EventCallback` runs [`asyncio.Event()`](https://github.com/wialon/gmqtt/blob/50bc08300d858c6e28d4975a989be578954bf7bd/gmqtt/mqtt/handler.py#L71).
5. `asyncio.Event()` tries to get the current event loop internally and then the event is created on this event loop.
6. `gunicorn` runs and `uvicorn` creates its own event loop. 
7. `MQTTClient` runs `await self._connected.wait()` at some point, and this operation runs on a different loop than the one `uvicorn` created.

In any case, possible solutions:
- Add `FastMQTT` initialization on a `startup` event.
- Ask for the `fastapi-mqtt` maintainer to change the place where `2` is done (it should be on their startup event).

**Comment by ruem2802** [comment]
@Kludex , I created start.sh:
#!/bin/bash
# Start Gunicorn processes
echo Starting Gunicorn.
exec gunicorn --reload server:app   -k uvicorn.workers.UvicornWorker  --bind 0.0.0.0:8000 --workers=9

And created run.py:
import subprocess
if __name__ == '__main__':
    subprocess.call(['sh', './start.sh'])

But it failed

**Comment by chengs** [comment]
any solutions?
